test/e2e/upgrade_test.go

```diff
@@ -241,3 +299,154 @@ func (tc *testContext) testWorkloadsAvailable(t *testing.T) {
 		})
 	assert.NoError(t, err)
 }
+
+// createParallelUpgradesCheckerJob deploys a Job that checks the number of nodes marked as upgrading and fails if the
+// number of parallel upgrades exceeds a threshold
+func (tc *testContext) createParallelUpgradesCheckerJob() (*batchv1.Job, error) {
+	checkerCommand := `
+		#!/bin/bash
+		set -euo pipefail
+		
+		# max number of parallel upgrades allowed, fixed to 1. Refer to controllers.MaxParallelUpgrades
+		export MAX_PARALLEL_UPGRADES=1
```

Comment:
Should we access `controllers.MaxParallelUpgrades` and Sprintf the checkerCommand so we can maintain this in 1 place?

Reply:
This was intentional to maintain an external verification of the controllers.MaxParallelUpgrades, given it has a fixed/constant value. So, this test verifies both the behavior and that the constant has the expected value.